packages/primitives/core/src/positioning/utils.ts

```diff
@@ -1,25 +1,52 @@
 import { ConnectedPosition, ConnectionPositionPair } from '@angular/cdk/overlay';
-import { POPOVER_POSITIONS } from './popover.constants';
+import { RDX_POSITIONS } from './constants';
 import {
+    RdxAlign,
     RdxAllPossibleConnectedPositions,
     RdxArrowPositionParams,
-    RdxPopoverAlign,
-    RdxPopoverSide,
+    RdxSide,
     RdxSideAndAlign,
     RdxSideAndAlignOffsets
-} from './popover.types';
+} from './types';
+
+export function getContentPosition(
+    sideAndAlignWithOffsets: RdxSideAndAlign & RdxSideAndAlignOffsets
+): ConnectedPosition {
+    const { side, align, sideOffset, alignOffset } = sideAndAlignWithOffsets;
+    const position = {
+        ...(RDX_POSITIONS[side]?.[align] ?? RDX_POSITIONS[RdxSide.Top][RdxAlign.Center])
+    };
+    if (sideOffset || alignOffset) {
+        if ([RdxSide.Top, RdxSide.Bottom].includes(side)) {
+            if (sideOffset) {
+                position.offsetY = side === RdxSide.Top ? -sideOffset : sideOffset;
+            }
+            if (alignOffset) {
+                position.offsetX = alignOffset;
+            }
+        } else {
+            if (sideOffset) {
+                position.offsetX = side === RdxSide.Left ? -sideOffset : sideOffset;
+            }
+            if (alignOffset) {
+                position.offsetY = alignOffset;
+            }
+        }
+    }
+    return position;
+}
 
 let allPossibleConnectedPositions: RdxAllPossibleConnectedPositions;
 export function getAllPossibleConnectedPositions() {
```

Comment:
```
let allPossibleConnectedPositions: RdxAllPossibleConnectedPositions | null = null;

getAllPossibleConnectedPositions(): {
    if (allPossibleConnectedPositions === null) {
        allPossibleConnectedPositions = new Map();

        for (const [side, aligns] of Object.entries(RDX_POSITIONS)) {
            if (!aligns) continue;
            for (const [align, position] of Object.entries(aligns)) {
                allPossibleConnectedPositions.set(`${side}|${align}`, position);
            }
        }
    }

    return allPossibleConnectedPositions;
}
```

The **current** implementation used `Object.keys`, which required additional lookups for each key to access its corresponding value.

The **refactored** implementation leverages `Object.entries`, which provides both key and value directly in a single iteration:
```typescript
for (const [side, aligns] of Object.entries(RDX_POSITIONS)) {
    for (const [align, position] of Object.entries(aligns)) {
        allPossibleConnectedPositions.set(`${side}|${align}`, position);
    }
}
```

**Impact**
- **Complexity**: No changes in the algorithmic complexity (still O(n)O(n)).
- improved **Performance**: fewer object property lookups reduce execution time, especially with large datasets.
- direct access to keys and values makes the code easier to understand and maintain.

Reply:
Great catch! Thanks @pimenovoleg 🚀